api_test.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2017 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keysync
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"net/http/httptest"
+	"net/url"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/square/go-sq-metrics"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestApiSyncAllAndSyncClientSuccess(t *testing.T) {
+	groupFile = "fixtures/ownership/group"
+	defer func() { groupFile = "/etc/group" }()
+
+	passwdFile = "fixtures/ownership/passwd"
+	defer func() { passwdFile = "/etc/passwd" }()
+
+	port := uint16(4444) // Shutting down the APIServer at the end of the test would require changing the method to return a pointer to the server
+
+	server := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Good point; I'll make another PR refactoring the tests to reduce reused code like this.  It's a bit tricky to factor this out since I want to defer closing the server as soon as possible, but can't defer closing inside the helper function; I could factor out some parts of this, though.